Tests/MapboxMapsTests/Foundation/Integration Tests/Map/MapViewRenderedSnapshotIntegrationTests.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import XCTest
+@testable @_spi(Experimental) import MapboxMaps
+
+final class MapViewRenderedSnapshotIntegrationTests: MapViewIntegrationTestCase {
+
+    func testLoadStyleAndTakeSnapshotSucceeds() {
+
+        guard let style = style else {
+            XCTFail("Should have a valid Style object")
+            return
+        }
+
+        let expectation1 = self.expectation(description: "Wait for style to load")
+        expectation1.expectedFulfillmentCount = 1
+
+        let expectation2 = self.expectation(description: "Wait for snapshot to be taken")
+        expectation2.expectedFulfillmentCount = 1
+
+        didFinishLoadingStyle = { _ in
+            expectation1.fulfill()
+        }
+
+        didBecomeIdle = { [weak self] _ in
+            guard let mapView = self?.mapView else {
+                XCTFail("Mapview must exist.")
+                return
+            }
+
+            do {
+                let image = try mapView.snapshot().get()
+                XCTAssertNotNil(image)
+                expectation2.fulfill()
+            } catch {
+                XCTFail("Snapshot failed with error: \(error)")
+            }
+        }
+
+        style.uri = .dark
+        wait(for: [expectation1, expectation2], timeout: 10)
+    }
+
+    func testSnapshotFailsDueToNoMetalView() {
+
+        guard let style = style else {
+            XCTFail("Should have a valid Style object")
+            return
+        }
+
+        let expectation1 = self.expectation(description: "Wait for style to load")
+        expectation1.expectedFulfillmentCount = 1
+
+        let expectation2 = self.expectation(description: "Wait for snapshot to fail")
+        expectation2.expectedFulfillmentCount = 1
+
```

Comment:
Same comments as above. 
`expectation1` -> `waitForStyleExpectation`
`expectation2` -> `snapshotFailedExpectation`

Also, remove lines 51 and 54. It is 1 by default